libclc/ptx-nvidiacl/libspirv/atomic/atomic_helpers.h

```diff
@@ -72,62 +72,72 @@ _CLC_OVERLOAD _CLC_DECL void __spirv_MemoryBarrier(unsigned int, unsigned int);
-      if (__clc_nvvm_reflect_arch() >= 700) {                                                                                \
-        __CLC_NVVM_ATOMIC_IMPL_ORDER(TYPE, TYPE_NV, TYPE_MANGLED_NV, OP,                                                     \
-                                     ADDR_SPACE, ADDR_SPACE_NV, _release)                                                    \
-      } else {                                                                                                               \
-        __spirv_MemoryBarrier(scope, Release);                                                                               \
-        __CLC_NVVM_ATOMIC_IMPL_ORDER(TYPE, TYPE_NV, TYPE_MANGLED_NV, OP,                                                     \
-                                     ADDR_SPACE, ADDR_SPACE_NV, )                                                            \
-      }                                                                                                                      \
-      break;                                                                                                                 \
-    case AcquireRelease:                                                                                                     \
-      if (__clc_nvvm_reflect_arch() >= 700) {                                                                                \
-        __CLC_NVVM_ATOMIC_IMPL_ORDER(TYPE, TYPE_NV, TYPE_MANGLED_NV, OP,                                                     \
-                                     ADDR_SPACE, ADDR_SPACE_NV, _acq_rel)                                                    \
-      } else {                                                                                                               \
-        __spirv_MemoryBarrier(scope, Release);                                                                               \
-        __CLC_NVVM_ATOMIC_IMPL_ACQUIRE_FENCE(TYPE, TYPE_NV, TYPE_MANGLED_NV,                                                 \
-                                             OP, ADDR_SPACE, ADDR_SPACE_NV)                                                  \
-      }                                                                                                                      \
-      break;                                                                                                                 \
-    }                                                                                                                        \
-    __builtin_trap();                                                                                                        \
-    __builtin_unreachable();                                                                                                 \
+#define __CLC_NVVM_ATOMIC_IMPL(TYPE, TYPE_MANGLED, TYPE_NV, TYPE_MANGLED_NV,   \
```

Comment:
Is this macro in use anywhere? Looks to me like all the uses were replaced with `__CLC_NVVM_ATOMIC_IMPL_SUBSTITUTION`. If that's the case, maybe we can keep the name instead of the new `__CLC_NVVM_ATOMIC_IMPL_SUBSTITUTION`?